src/stdlib/controlflow/bind.verona

```diff
@@ -3,24 +3,12 @@
 
 // `~` implements explicit partial application.
 // f ~x rewrites to { y => f(x, y) }
-~[T, U, V, R: U->V](f: R~>((T, U)->V), x: R~>T): R
+~[T, U..., V, R: U...->V](f: R~>((T, U...)->V), x: R~>T): R
 {
-  { y => f(x, y) }
+  { y: U... => f(x, y) }
 }
 
-|>[T, U](x: T, y: T->U): U
+|>[T..., U](x: T..., y: T...->U): U
```

Comment:
Does this means that `x` (being a type list) can still partially apply?

Ex:
```
f(a: A, b: B, c: C);
var p: C;
let partialC = f ~p; // creates partialC(a: A, b: B);
// here, does |>(x: T..., y: T...->U): U "partially applies"?
```

Reply:
In your example:

```
f: (A, B, C)->None
```

so you can't do `f ~p` because `p: C`. You could have `p: A`, for example:

```
var p: A = ...;
let partialF = f ~p; // partialF: (B, C)->None
```

Then, you could use `|>` like this:

```
var b: B = ...;
var c: C = ...;

(b, c) |> partialF;
// this gives partialF(b, c), which will call f(p, b, c)
```

Reply:
Right, I was unsure about which way partial application goes (my first example was with `a` but I got confused).

Ok, that looks like what I was expecting, thanks!